src/main/java/pl/zaradny/springApp/domain/Product.java

```diff
@@ -34,18 +42,22 @@ public Price getPrice() {
         return price;
     }
 
-    public Description getDescription() {
-        return description;
+    public Optional<Description> getDescription() {
+        return Optional.ofNullable(description);
     }
 
-    public Image getImage() {
-        return image;
+    public Optional<Image> getImage() {
+        return Optional.ofNullable(image);
     }
 
     public LocalDateTime getCreatedAt() {
         return createdAt;
     }
 
+    public Optional<Set<Tag>> getTags() {
```

Comment:
i tutaj mamy gwiazdę programu, to znaczy wewnątrz twojej aplikacji możesz wywołać getTags a następnie na referencji którą otrzymałeś dodać element do set-a i już problem gotowy, obiekt domenowy nam wtedy mutuje ;)

na szybko przynajmniej kilka przykładów skąd wziąć niemutowalny set: https://www.techiedelight.com/mutable-unmodifiable-immutable-empty-set-java/

najlepiej zadbać o to na poziomie konstruktora i w momencie przypisywania do pola `this.tags = tags;` zrobić: `this.tags = ImmutableSet.copyOf(tags);` i już masz pewność że w żadnym momencie pole nie zostanie zmutowane